torch/testing/_internal/distributed/rpc/dist_optimizer_test.py

```diff
@@ -18,11 +18,13 @@ class MyModule:
     lock = threading.Lock()
 
     def __init__(self):
-        # avoid race where 2 modules could be initialized
-        # concurrently thus changing the order random numbers are drawn.
-        with MyModule.lock:
-            torch.manual_seed(0)
-            self.w = torch.rand((3, 3), requires_grad=True)
+        # cannot directly use torch.manual_seed(0) as all threads share the same
+        # default generator. The race from multiple RPC threads could mess up
+        # the draw order from the default RNG instance, leading to
+        # non-deterministic behavior. Hence, create a dedicated RNG here.
+        g_cpu = torch.Generator()
```

Comment:
Do we no longer need the global lock that is removed?

Reply:
IIUC, we no longer need it as we will create a new RNG instance every time we need it.